mobile/library/java/io/envoyproxy/envoymobile/engine/EnvoyEngineImpl.java

```diff
@@ -93,16 +89,20 @@ public int runWithTemplate(String configurationYAML, EnvoyConfiguration envoyCon
       JniLibrary.registerKeyValueStore(entry.getKey(),
                                        new JvmKeyValueStoreContext(entry.getValue()));
     }
+  }
 
-    return runWithResolvedYAML(
-        envoyConfiguration.resolveTemplate(
-            configurationYAML, JniLibrary.platformFilterTemplate(),
-            JniLibrary.nativeFilterTemplate(), JniLibrary.altProtocolCacheFilterInsert(),
-            JniLibrary.gzipConfigInsert(), JniLibrary.brotliConfigInsert(),
-            JniLibrary.socketTagConfigInsert(), JniLibrary.persistentDNSCacheConfigInsert(),
-            JniLibrary.certValidationTemplate(
-                envoyConfiguration.enablePlatformCertificatesValidation)),
-        logLevel);
+  /**
+   * Run the Envoy engine with the provided yaml string and log level.
+   *
+   * This does not perform registration, and performRegistration may need to be called first.
+   *
+   * @param configurationYAML The configuration yaml with which to start Envoy.
+   * @param logLevel          The log level to use when starting Envoy.
+   * @return A status indicating if the action was successful.
```

Comment:
Not new to this PR, but it's ambiguous what status values indicate success. Is zero success like unix exit codes or is one success like boolean 0/false and 1/true?

Reply:
I wonder if we should switch these ints over to the Java enum EnvoyStatus?

Reply:
Probably a good idea

Reply:
Yes, out of scope here.

Reply:
100%